plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergStatistics.java

```diff
@@ -0,0 +1,328 @@
+                    updateMinMaxStats(
+                            id,
+                            trinoType,
+                            lowerBound,
+                            upperBound,
+                            nullCount,
+                            dataFile.recordCount());
+                    updateNullCountStats(
+                            id,
+                            nullCount);
+                }
+            }
+        }
+
+        public IcebergStatistics build()
+        {
+            ImmutableMap.Builder<Integer, Object> minValues = ImmutableMap.builder();
+            ImmutableMap.Builder<Integer, Object> maxValues = ImmutableMap.builder();
+
+            columnStatistics.forEach((fieldId, statistics) -> {
+                statistics.getMin().ifPresent(min -> minValues.put(fieldId, min));
+                statistics.getMax().ifPresent(max -> maxValues.put(fieldId, max));
+            });
+
+            Map<Integer, Long> nullCounts = this.nullCounts.entrySet().stream()
+                    .filter(entry -> entry.getValue().isPresent())
+                    .collect(toImmutableMap(Map.Entry::getKey, entry -> entry.getValue().orElseThrow()));
+
+            return new IcebergStatistics(
+                    recordCount,
+                    fileCount,
+                    size,
+                    minValues.build(),
+                    maxValues.build(),
+                    nullCounts,
+                    ImmutableMap.copyOf(columnSizes));
+        }
+
+        private void updateNullCountStats(int id, Optional<Long> nullCount)
+        {
+            // If one file is missing nullCounts for a column, invalidate the estimate
+            if (nullCounts.containsKey(id) && nullCounts.get(id).isEmpty()) {
+                return;
+            }
+
+            if (nullCount.isPresent()) {
+                nullCounts.merge(id, nullCount, (existingCount, newCount) -> Optional.of(existingCount.get() + newCount.get()));
+            }
+            else {
+                nullCounts.put(id, Optional.empty());
+            }
+        }
+
+        private void updateMinMaxStats(
+                int id,
+                io.trino.spi.type.Type type,
+                Object lowerBound,
+                Object upperBound,
+                Optional<Long> nullCount,
+                long recordCount)
+        {
+            // If this column is only nulls for this file, don't update or invalidate min/max statistics
+            if (nullCount.isEmpty() || nullCount.get() != recordCount) {
+                columnStatistics.putIfAbsent(id, new ColumnStatistics(type, typeManager, lowerBound, upperBound));
+                columnStatistics.get(id).updateMinMax(lowerBound, upperBound);
+            }
+        }
+
+        private static Map<Integer, Object> convertBounds(Map<Integer, Type.PrimitiveType> idToTypeMapping, Map<Integer, ByteBuffer> idToMetricMap)
```

Comment:
make non-static and drop `idToTypeMapping` arg

Reply:
document what Object it returns (Iceberg representation, Trino representation, something else)